tests/metaworld/benchmarks/test_mt10.py

```diff
@@ -10,6 +10,10 @@ def test_augment_observation():
     env = MT10()
     for i in range(env.num_tasks):
         env.set_task(i)
+        # Right now, env.reset() calls env.active_env.set_task() under the hood
+        # This is necessary before stepping, and calling the method directly
```

Comment:
does it call set_task on all of the environments or just the active one? multitaskenv keeps a bunch of environments and only one is active at a time, but in this new style it will need to call set_task on each of them.

Reply:
`MultiClassMultiTaskEnv` has a `set_task` whose input is sometimes an integer (task ID) and sometimes a dict containing a task ID and a goal. This function only touches the active environment insofar as it sets its `self.goal` to the goal from the task dict. (Side note -- that may be causing its own set of inconsistencies, as `self.goal` only gets applied to `self._state_goal` if `reset()` is called before `step()`, and we don't always do that in tests)

Anyway, my first thought was to call `active_env.set_task()` (the new function) inside of `MultiClassMultiTaskEnv.set_task()`, but that didn't always work. It seemed like there were multiple reasons for this, the most common being that tests sometimes modify `_active_task` manually rather than calling `MultiClassMultiTaskEnv.set_task()`. When they do that, `active_env` gets updated to an env that hasn't yet had `set_task()` called.

My simple fix was to call `active_env.set_task()` every time `step` or `reset` are called on MultiClassMultiTaskEnv. This ensures that the active environment is always ready to go.

I'm sure there's a better place to put this logic, but I'm not sure where.

Reply:
I think that deleting MultiClassMultiTaskEnv is an aim of Avnish's API changes, so I'm not really concerned with what goes on here.